src/metricscollector/collector/apppoller_test.go

```diff
@@ -0,0 +1,191 @@
+package collector_test
+
+import (
+	"metricscollector/cf"
+	. "metricscollector/collector"
+	"metricscollector/collector/fakes"
+	"metricscollector/metrics"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/cloudfoundry/sonde-go/events"
+	"github.com/gogo/protobuf/proto"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"errors"
+	"time"
+)
+
+var _ = Describe("Apppoller", func() {
+
+	var (
+		cfc      *fakes.FakeCfClient
+		noaa     *fakes.FakeNoaaConsumer
+		database *fakes.FakeDB
+		poller   *AppPoller
+	)
+
+	BeforeEach(func() {
+		cfc = &fakes.FakeCfClient{}
+		noaa = &fakes.FakeNoaaConsumer{}
+		database = &fakes.FakeDB{}
+		logger := lager.NewLogger("apppoller-test")
+
+		poller = NewAppPoller(logger, "test-app-id", TestPollInterval*time.Second, cfc, noaa, database)
+	})
+
+	Describe("Start", func() {
+
+		JustBeforeEach(func() {
+			poller.Start()
```

Comment:
This is wrong.
You start the poller here.
You stop the poller in the AfterEach.